### DISCUSSION TITLE
Comet orbital elements from MPC query

### DISCUSSION THREAD

**Discussion by jurezakrajsek** [skyfielders/python-skyfield#675] [Show and tell]
The SkyField library is fantastic. 
I use it now for lots of calculation regarding comets on my COBS project.
One particular thing i found useful was an additional function to retrieve the latest orbital elements in MPC 1-line format for a specific comet as a directly query from MPC website.

```
# Get comet orbital elements from MPC query
def get_mpc_ephemeris_response(comet_name):
    # Split if full comet name is provided
    comet_name = comet_name.split('(')[0].strip()

    # A CGI query is produced
    query = {
        'ty': 'e', 'TextArea': comet_name, 'd': '', 'l': '', 'i': '', 'u': 'd', 'uto': '0', 'c': '106', 'long': '',
        'lat': '', 'alt': '', 'raty': 'a', 's': 's', 'm': 'm', 'igd': '', 'ibh': '', 'fp': '', 'oed': '', 'e': -1,
        'res': 'y', 'resoc': '', 'tit': '', 'bu': '', 'ch': 'c', 'ce': 'f', 'js': 'f'
    }

    # Send the query to the MPC site
    data = urllib.parse.urlencode(query).encode('UTF-8')
    req = urllib.request.Request('https://cgi.minorplanetcenter.net/cgi-bin/mpeph2.cgi', data)
    response = urllib.request.urlopen(req)
    orb = response

    return orb
```

The orbit is obtained as a bytes line that can be then read directly by the mpc.load_comets_dataframe_slow to create a comet DataFrame object.

```
comet_name = 'C/2019 L3 (ATLAS)'
orbit = get_mpc_ephemeris_response(comet_name)
comet = mpc.load_comets_dataframe_slow(orb)
print(comet)
```

Clear skies
Jure

**Comment by brandon-rhodes** [comment]
Thank you for this example code! I should be able to look at it sometime early this month, and I'll follow up here with another comment when I have the chance.

**Reply by jurezakrajsek** [reply]
Hi @brandon-rhodes 
Any updates on this. Could this be included directly into SkyField?

One could also consider using the astroquery option to retrieve the comet orbital elements
https://astroquery.readthedocs.io/en/latest/mpc/mpc.html
